src/com/koxudaxi/poetry/PyAddExistingPoetryEnvPanel.kt

```diff
@@ -55,20 +58,32 @@ class PyAddExistingPoetryEnvPanel(private val project: Project?,
                 .panel
         add(formPanel, BorderLayout.NORTH)
         addInterpretersAsync(sdkComboBox) {
-            detectPoetryEnvs(module, existingSdks, context, project?.basePath ?: newProjectPath)
+            val existingSdkPaths = sdkHomes(existingSdks)
+            val moduleSdks = allModules(project).parallelStream().flatMap { module ->
+                val sdks = detectPoetryEnvs(module, existingSdkPaths, module.basePath)
+                        .filterNot { it.isAssociatedWithAnotherModule(module) }
+                sdks.forEach { sdkToModule.putIfAbsent(it.name, module) }
+                sdks.stream()
+            }.toList()
+            val rootSdks = detectPoetryEnvs(module, existingSdkPaths, project?.basePath ?: newProjectPath)
                     .filterNot { it.isAssociatedWithAnotherModule(module) }
+            val moduleSdkPaths = moduleSdks.map { it.name }.toSet()
+            val sdks = rootSdks.filterNot { moduleSdkPaths.contains(it.name) } + moduleSdks
```

Comment:
Ensure we only have one entry for the project if there's a module at the same level:
![Uploading Add_Python_Interpreter_and_Project_Structure.png…]()

Reply:
Ensure that associated sdks are not in the dropdown:
<img width="953" alt="Project_Structure_and_data-mesh" src="https://user-images.githubusercontent.com/527952/116505160-34d00a00-a888-11eb-969f-12ad97a81cae.png">
<img width="958" alt="Add_Python_Interpreter_and_Project_Structure_and_data-mesh" src="https://user-images.githubusercontent.com/527952/116505210-50d3ab80-a888-11eb-9003-4bc43c391a92.png">